modules/internal/ChapelArray.chpl

```diff
@@ -1558,37 +1558,49 @@ module ChapelArray {
     /* Return true if this domain is a subset of ``super``. Otherwise
        returns false. */
     proc isSubset(super : domain) {
-      if !isAssociativeDom(this) {
-        if isRectangularDom(this) then
-          compilerError("isSubset not supported on rectangular domains");
-        else if isOpaqueDom(this) then
+      if !(isAssociativeDom(this) || isRectangularDom(this)) {
+        if isOpaqueDom(this) then
           compilerError("isSubset not supported on opaque domains");
         else if isSparseDom(this) then
           compilerError("isSubset not supported on sparse domains");
         else
           compilerError("isSubset not supported on this domain type");
       }
       if super.type != this.type then
-        compilerError("isSubset called with different associative domain types");
+        if isRectangularDom(this) {
+          if super.dims().size != this.dims().size then
+            compilerError("isSubset called with different rectangular domain types");
+        } else
+          compilerError("isSubset called with different associative domain types");
+
+      if isRectangularDom(this) then
+        return && reduce for i in 1..this.dims().size do
```

Comment:
I'm still expecting a for loop instead of the && reduce, e.g.
``` chapel
var contains = true;
for i in 1..this.dims().size {
  contains && = super.dims()[i].contains(this.dims()[i]);
}
```

Reply:
Oh, I'm sorry, I misunderstood it the last time. On it.

Reply:
Done.
Also added a break statement to optimize performance.